### PR TITLE
[decrypt_all] Check if file needs to decrypted or not for speed up large oc setups.

### PR DESCRIPTION
**Pull request by chriseqipe**

### CODE REVIEW

File: lib/private/encryption/decryptall.php [hunk 1]
```diff
@@ -213,9 +214,14 @@ protected function decryptUsersFiles($uid, ProgressBar $progress, $userCount) {
 					try {
 						$progress->setMessage("decrypt files for user $userCount: $path");
 						$progress->advance();
-						if ($this->decryptFile($path) === false) {
+						if ($file->getData()['encrypted'] == 0) {
```
Comment (LukasReschke) [inline lib/private/encryption/decryptall.php]: Please always use `===`. If you want to typecast here use `(int)$file->getData()['encrypted']`.

**Comment by chriseqipe** [comment]
Thx for the hint. In fact $file->getData()['encrypted'] comes as "0" String. I decided to cast it to int.

**Comment by PVince81** [comment]
@schiesbn what do you think ?

**Comment by MorrisJobke** [comment]
@LukasReschke @schiesbn Please review ;)

File: lib/private/encryption/decryptall.php [hunk 2]
```diff
@@ -213,9 +214,14 @@ protected function decryptUsersFiles($uid, ProgressBar $progress, $userCount) {
 					try {
 						$progress->setMessage("decrypt files for user $userCount: $path");
 						$progress->advance();
-						if ($this->decryptFile($path) === false) {
+						if ((int)$file->getData()['encrypted'] === 0) {
```
Comment (schiesbn) [inline lib/private/encryption/decryptall.php]: please call  `$file->isEncrypted();` instead.

Otherwise it looks good, thanks a lot for your pull request!

**Comment by PVince81** [comment]
@chriseqipe something looks wrong with your commit list.

Can you redo a PR ? Best is to create a branch in your own repo, put the changes on the branch and submit the PR from there.

**Comment by PVince81** [comment]
Something went wrong with this PR, see commit list.

Please resubmit.

**Comment by PVince81** [comment]
Ok, I have spotted the commits, I'll make a new PR and we'll continue from there.

**Comment by PVince81** [comment]
I cherry picked the commits into a new PR: https://github.com/owncloud/core/pull/24628
Let's continue there

**Pull request closed without merging**